### PR TITLE
financial transaction category creation

### PR DESCRIPTION
**Pull request by zetxdev**

### CODE REVIEW

**Review by dawciobiel — COMMENTED** [review]
I removed two no need files.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryCreateDTO;
+import pl.byczazagroda.trackexpensesappbackend.dto.FinancialTransactionCategoryDTO;
+import pl.byczazagroda.trackexpensesappbackend.mapper.FinancialTransactionCategoryModelMapper;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionCategory;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+@Service
+@RequiredArgsConstructor
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:13]: It's a good idea to add annotations in alphabetical order. This way, in every class or class method, it always looks the same.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+@Validated
+public class FinancialTransactionCategoryServiceImpl implements FinancialTransactionCategoryService {
+
+    private final FinancialTransactionCategoryRepository financialTransactionCategoryRepository;
+    private final FinancialTransactionCategoryModelMapper financialTransactionCategoryModelMapper;
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:18]: One blank line required between declaration of class fields.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java [hunk 3]
```diff
@@ -26,39 +27,42 @@
...
         List<FinancialTransactionCategoryDTO> financialTransactionCategoryDTOList = null;
         return new ResponseEntity<>(financialTransactionCategoryDTOList, HttpStatus.OK);
     }
 
     @PostMapping()
-    public ResponseEntity<FinancialTransactionCategoryDTO> createFinancialTransactionCategory(@Valid @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO){
-        //TODO Necessary code implementation
-        FinancialTransactionCategoryDTO financialTransactionCategoryDTO = null;
+    public ResponseEntity<FinancialTransactionCategoryDTO> createFinancialTransactionCategory(
+            @Valid @RequestBody FinancialTransactionCategoryCreateDTO financialTransactionCategoryCreateDTO) {
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionCategoryController.java:50]: When you are using annotation `@Valid` then `@Validated` on the class is necessary as well.
Anyway please remove `@Valid` from this method and add it to the service method instead. As well please add `@Validated` to the service class.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    @Override
+    public FinancialTransactionCategoryDTO createFinancialTransactionCategory(
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:21]: Please add the `@Valid` annotation to this method.

Comment (dawciobiel) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImpl.java:21]: The shorter version of Your code:

```java
public FinancialTransactionCategoryDTO createFinancialTransactionCategory(@Valid FinancialTransactionCategoryCreateDTO dto) {
			
	FinancialTransactionCategory entityToSave = new FinancialTransactionCategory(dto.name(), dto.type());
	FinancialTransactionCategory savedEntity = financialTransactionCategoryRepository.save(entityToSave);
			
	return financialTransactionCategoryModelMapper.mapFinancialTransactionCategoryEntityToFinancialTransactionCategoryDTO(savedEntity);
}
```

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionCategoryRepository;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class FinancialTransactionCategoryServiceImplTest {
+
+    public static final long ID_1L = 1L;
```
Comment (dawciobiel) [inline src/test/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionCategoryServiceImplTest.java:23]: A blank line between class fields is recommended

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

**Review by dawciobiel — APPROVED** [review]
Looks good for me

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java [hunk 6]
```diff
@@ -8,7 +8,6 @@ public enum ErrorCode {
     W004("W004", "WALLETS_LIST_LIKE_NAME_NOT_FOUND_EXC_MSG", 404),
 
     FT001("FT001", "FINANCIAL_TRANSACTION_NOT_FOUND", 404),
-
```
Comment (odzio33) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/exception/ErrorCode.java:11]: Please restore this empty line becouse it was between two diff error codes

**Review by odzio33 — CHANGES_REQUESTED** [review]
Changes require

**Review by dawciobiel — DISMISSED** [review]
Looks good to me.

**Review by odzio33 — DISMISSED** [review]
Looks good for me

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by dawciobiel — DISMISSED** [review]
Looks good for me.

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by dawciobiel — CHANGES_REQUESTED** [review]
changes required.

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by dawciobiel — APPROVED** [review]
Looks good to me.

**Pull request merged**